lib/ansible/modules/cloud/azure/azure_rm_policydefinition.py

```diff
@@ -0,0 +1,398 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Fan Qiu, (fanqiu@microsoft.com)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_policydefinition
+version_added: "2.9"
+short_description: Manage Azure Policy Definition.
+description:
+    - Create, update and delete instance of Azure policy definition.
+
+options:
+    name:
+        description:
+            - Name of the policy definition
+        required: True
+    management_group:
+        description:
+            - The ID of the management group.
+    policy_type:
+        description:
+            - The type of policy definition.
+            - Possible value is Custom.
+        choices:
+            - custom
+    mode:
+        description:
+            - The policy definition mode.
+            - Possible values are Indexed, and All.
+        choices:
+            - indexed
+            - all
+    display_name:
+        description:
+            - The display name of the policy definition.
+    description:
+        description:
+            - The policy definition description.
+    policy_rule:
+        description:
+            - The policy rule.
+            - Policy rules in JSON format, or a path or url to a file containing JSON rules.
+            - Refer U(https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/2018-05-01/policydefinitions) for more details.
+            - This link U(https://docs.microsoft.com/en-us/azure/governance/policy/concepts/definition-structure) can be helpful.
+    parameters:
+        description:
+            - Required if a parameter is used in policy rule.
+            - JSON formatted string or a path to a file or url with parameter definitions.
+            - Refer U(https://docs.microsoft.com/en-us/azure/templates/microsoft.authorization/2018-05-01/policydefinitions) for more details.
+    metadata:
+        description:
+            - The policy definition metadata.
+            - Metadata in space-separated key=value pairs.
```

Comment:
I add the link and explanation for this parameters. It defines [subproperties primarily used by the Azure portal to display user-friendly information](https://docs.microsoft.com/en-us/azure/governance/policy/concepts/definition-structure#parameter-properties). Although there are some examples for this parameter. During my test, it can set any string:string information into the metadata.